CONTRIBUTING.md

```diff
@@ -0,0 +1,241 @@
+
+## Operating System
+We develop on macOS and all the scripts are written for that. However, we have contributors who develop on Windows and Linux. We try to make the setup as easy as 
+possible for all operating systems, but if you run into any issues, please let us know.
+
+**WARNING:** Developing on Windows can be a bit more challenging than on macOS or Linux. Tests take much longer to start and run and the configuration is quite finicky. We
+**strongly** recommend using the GitHub bash shell for development on Windows.
+
+## 🚀 Getting Started (Python)
+
+First, let's make sure we have everything we need to get started.
+### Prerequisites (macOS)
+
+##### Ensure Python3.11 is installed
+```bash
+brew install python@3.11
+```
+
+##### [Python Poetry](https://python-poetry.org/docs/ "Official poetry installation guide")
+In this walkthrough we use will use Poetry to manage our venv.
+```bash
+pipx install poetry
+```
+
+##### [Install MongoDB](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-os-x/)
+```bash
+brew tap mongodb/brew
+brew install mongodb-community@5.0
+```
+
+
+##### [GitHub CLI](https://cli.github.com/) (optional)
+```bash
+brew install gh
+gh auth login -h GitHub.com -w -p https
+```
+
+### Prerequisites (linux)
+
+##### Ensure Python3.11 is installed
+```bash
+apt install python3
+```
+
+##### [Python Poetry](https://python-poetry.org/docs/ "Official poetry installation guide")
+```bash
+pipx install poetry
+```
+
+##### [Install MongoDB](https://www.mongodb.com/docs/manual/tutorial/install-mongodb-on-debian/)
+*follow the instructions for your distribution*
+
+##### [GitHub CLI](https://cli.github.com/) (optional)
+```bash
+apt install gh
+gh auth login -h GitHub.com -w -p https
+```
+
+### Setup Eidolon
+
+Now that we have all the prerequisites, let's fork the Eidolon monorepo and get started!
+
+##### [Fork Eidlon monorepo](https://github.com/eidolon-ai/eidolon/fork)
+```bash
+gh repo fork eidolon-ai/eidolon --clone=true
+cd eidolon
+```
+
+##### [OpenAI API Key](https://platform.openai.com/account/api-keys "Create an OpenAI key")
+Visit the open AI site and create an API key. You will need to set this as an environment variable in the .env file.
+
+From the root directory of Eidolon, create a .env file and add your OpenAI key.
+```bash
+echo "OPENAI_API_KEY=<YOUR OPENAI>" > .env
+```
+
+Other optional environment variables are:
+```bash
+echo "MONGO_CONNECTION_STR=mongodb://localhost:27017/?directConnection=true" >> .env
+echo "MONGO_DATABASE_NAME=eidolon" >> .env
+echo "MISTRAL_API_KEY=<YOUR MISTRAL_API_KEY>" >> .env
+echo "ANTHROPIC_API_KEY=<YOUR ANTHROPIC_API_KEY>" >> .env
+```
+
+### Build and Run Eidolon
+There are 3 main components that need to be built and run: the client libraries, the server, and the examples:
+
+```bash
+cd client/python
+poetry shell
+poetry install
+deactivate
+```
```

Comment:
I think we can remove these scripts and replace it with a word about using poetry to manage dependencies. If we choose to keep the scripts, you do not need to have to activate/deactivate the shell each time.

More important is a description of the repository, what each project is, how to run tests, and probably a word on the cassettes.